ray-on-gke/user/kuberay-tpu-webhook/main.go

```diff
@@ -0,0 +1,242 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	ray "github.com/ray-project/kuberay/ray-operator/apis/ray/v1"
+	admissionv1 "k8s.io/api/admission/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+)
+
+// our representation of a pod slice
+// not necessarily true that worker group scheduled on 1 slice
+type Slice struct {
+	rayClusterName	string
+	groupName	string
+}
+
+var (
+	tpuResourceName = corev1.ResourceName("google.com/tpu")
+	certPath = "/etc/kuberay-tpu-webhook/tls/tls.crt"
+	keyPath = "/etc/kuberay-tpu-webhook/tls/tls.key"
+
+	// map of ray cluster names to # of workers created in the slice
+	sliceToWorkers map[Slice]int
+)
+
+// check if containers are requesting TPU resources
+func containerRequestingTPUs(containers ...corev1.Container) bool {
```

Comment:
Should we also check here if the request is for multi-host TPUs? Single-host should not be running this webhook.

Reply:
That sounds reasonable, I added in checks so that only workers requesting TPU resources with a multi-host topology will be mutated.